appcenter/derived_client.py

```diff
@@ -185,6 +185,11 @@ def get(self, url: str) -> requests.Response:
         """
 
         response = self.session.get(url)
+        
+        # Check if the response is HTML
+        if 'text/html' in response.headers['Content-Type']:
+            self.log.info("The response is HTML")
+            self.log.info(f"Response: {response.text}")
```

Comment:
1. Maybe we can check if the  content type is not `application/json`, any thoughts? @dalemyers 
2. We can also log other useful information for investigation, such as
```python
self.log.info((
    "Getting non json response: "
    f"method={response.request.method}, "
    + f"url={response.request.url}, "
    + f"status_code={response.status_code}, "
    + f"text={response.headers['Content-Type']}"
    + f"text={response.text}"
))
```

Reply:
I think @Cokile is correct. We want to throw an error if it's anything else. To make sure we don't break anything in the future, I'd have a parameter we check against. e.g.

```swift
def get(self, url: str, *, expected_content_type: str = "application/json") -> request.Response
```

Then have 

```swift
if response.headers.get("Content-Type") != expected_content_type:
    raise AppCenterHTTPException(response)
```

It's vital we raise an exception rather than logging as we have no idea what sort of data could potentially be in those responses. We need to let the consumer decide how to handle this rather than potentially leaking it by accident due to misconfigured logging.

Reply:
(Same for the other methods obviously)